powerstore/resource_protection_policy.go

```diff
@@ -3,130 +3,187 @@ package powerstore
+
+			"name": schema.StringAttribute{
 				Required:            true,
 				Description:         "The name of the protection policy.",
 				MarkdownDescription: "The name of the protection policy.",
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
 			},
-			"description": {
-				Type:                types.StringType,
-				Computed:            true,
+
+			"description": schema.StringAttribute{
 				Optional:            true,
+				Computed:            true,
 				Description:         "The description of the protection policy.",
 				MarkdownDescription: "The description of the protection policy.",
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
 			},
-			"type": {
-				Type:                types.StringType,
-				Computed:            true,
+
+			"type": schema.StringAttribute{
 				Optional:            true,
+				Computed:            true,
 				Description:         "The type of the protection policy.",
 				MarkdownDescription: "The type of the protection policy.",
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
 			},
-			"managed_by": {
-				Type:                types.StringType,
-				Computed:            true,
+
+			"is_read_only": schema.BoolAttribute{
 				Optional:            true,
-				Description:         "Entity that owns and manages this instance.",
-				MarkdownDescription: "Entity that owns and manages this instance.",
-			},
-			"managed_by_id": {
-				Type:                types.StringType,
 				Computed:            true,
-				Optional:            true,
-				Description:         "ID of the managing entity.",
-				MarkdownDescription: "ID of the managing entity.",
-			},
-			"is_read_only": {
-				Type:                types.BoolType,
-				Computed:            true,
-				Optional:            true,
 				Description:         "Indicates whether this policy can be modified.",
 				MarkdownDescription: "Indicates whether this policy can be modified.",
 			},
-			"is_replica": {
-				Type:                types.BoolType,
-				Computed:            true,
+
+			"snapshot_rule_ids": schema.SetAttribute{
+				ElementType:         types.StringType,
 				Optional:            true,
-				Description:         "Indicates if this is a replica of a policy.",
-				MarkdownDescription: "Indicates if this is a replica of a policy.",
-			},
-			"snapshot_rule_ids": {
-				Type:                types.SetType{ElemType: types.StringType},
 				Computed:            true,
-				Optional:            true,
-				Description:         "List of the snapshot_rule IDs that are associated with this policy.",
-				MarkdownDescription: "List of the snapshot_rule IDs that are associated with this policy.",
+				Description:         "List of the snapshot rule IDs that are associated with this policy.",
+				MarkdownDescription: "List of the snapshot rule IDs that are associated with this policy.",
+				Validators: []validator.Set{
+
```

Comment:
Is it validating the snapshot_rule_ids = [] or  [""] values also?

Reply:
It validates both `snapshot_rule_ids = []` & `snapshot_rule_ids = [""]`